src/stream/src/executor/aggregation/agg_impl/foldable.rs

```diff
@@ -649,4 +657,354 @@ mod tests {
+            0.05,
+            100,
+            false,
+        );
+    }
+
+    #[bench]
+    fn bench_foldable_min_agg_without_vis(b: &mut Bencher) {
+        bench_i64(
+            b,
+            Box::<TestStreamingMinAgg<I64Array>>::default(),
+            "min",
+            1024,
+            1.0,
+            100,
+            true,
+        );
+    }
+
+    #[bench]
+    fn bench_foldable_min_agg_vis_rate_0_75(b: &mut Bencher) {
+        bench_i64(
+            b,
+            Box::<TestStreamingMinAgg<I64Array>>::default(),
+            "min",
+            1024,
+            0.75,
+            100,
+            true,
+        );
+    }
+
+    #[bench]
+    fn bench_foldable_min_agg_vis_rate_0_5(b: &mut Bencher) {
+        bench_i64(
+            b,
+            Box::<TestStreamingMinAgg<I64Array>>::default(),
+            "min",
+            1024,
+            0.5,
+            100,
+            true,
+        );
+    }
+
+    #[bench]
+    fn bench_foldable_min_agg_vis_rate_0_25(b: &mut Bencher) {
+        bench_i64(
+            b,
+            Box::<TestStreamingMinAgg<I64Array>>::default(),
+            "min",
+            1024,
+            0.25,
+            100,
+            true,
+        );
+    }
+
+    #[bench]
+    fn bench_foldable_min_agg_vis_rate_0_05(b: &mut Bencher) {
+        bench_i64(
+            b,
+            Box::<TestStreamingMinAgg<I64Array>>::default(),
+            "min",
+            1024,
+            0.05,
+            100,
+            true,
+        );
+    }
+
+    #[bench]
+    fn bench_foldable_max_agg_without_vis(b: &mut Bencher) {
+        bench_i64(
+            b,
+            Box::<TestStreamingMaxAgg<I64Array>>::default(),
+            "max",
+            1024,
+            1.0,
+            100,
+            true,
+        );
+    }
+
+    #[bench]
+    fn bench_foldable_max_agg_vis_rate_0_75(b: &mut Bencher) {
+        bench_i64(
+            b,
+            Box::<TestStreamingMaxAgg<I64Array>>::default(),
+            "max",
+            1024,
+            0.75,
+            100,
+            true,
+        );
+    }
+
+    #[bench]
+    fn bench_foldable_max_agg_vis_rate_0_5(b: &mut Bencher) {
+        bench_i64(
+            b,
+            Box::<TestStreamingMaxAgg<I64Array>>::default(),
+            "max",
+            1024,
+            0.5,
+            100,
+            true,
+        );
+    }
+
+    #[bench]
+    fn bench_foldable_max_agg_vis_rate_0_25(b: &mut Bencher) {
+        bench_i64(
+            b,
+            Box::<TestStreamingMaxAgg<I64Array>>::default(),
+            "max",
+            1024,
+            0.25,
+            100,
+            true,
+        );
+    }
+
+    #[bench]
+    fn bench_foldable_max_agg_vis_rate_0_05(b: &mut Bencher) {
+        bench_i64(
+            b,
+            Box::<TestStreamingMaxAgg<I64Array>>::default(),
+            "max",
+            1024,
+            0.05,
+            100,
+            true,
+        );
+    }
```

Comment:
I think this is fine for now, but you could use a macro or function to generate all these benchmark cases to reduce LoC. ~250 lines are boilerplate.